packages/jaeger-ui/src/reducers/deep-dependency-graph.tsx

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { handleActions } from 'redux-actions';
+
+import { actionTypes } from '../actions/deep-dependency-graph';
+import { fetchDeepDependencyGraph } from '../actions/jaeger-api';
+import { fetchedState } from '../constants';
+import { ApiError } from '../types/api-error';
+import transformDdgData from '../model/ddg/transformDdgData';
+import {
+  stateKey,
+  TDdgActionMeta,
+  TDdgAddStylePayload,
+  TDdgClearStylePayload,
+  TDdgPayload,
+  TDdgState,
+  TDdgStateEntry,
+} from '../model/ddg/types';
+import guardReducer, { guardReducerWithMeta } from '../utils/guardReducer';
+
+export function addStyleState(state: TDdgState, { payload }: { payload: TDdgAddStylePayload }) {
+  const { service, operation, start, end, visibilityIndices, style } = payload;
+  const key = stateKey(service, operation, start, end);
```

Comment:
I don't think components that dispatch actions need to be concerned with how the key is made. Keeping it as args on the payload means we can refactor the redux state as we please.
For instance, if, for some reason, we need to "unflatten" the state, that would be easy to do in just the action and reducer files, rather than any and all consuming files.

Reply:
stateKey taking a kwarg cleans this up significantly